app/src/lib/git/core.ts

```diff
@@ -264,6 +264,12 @@ function getDescriptionForError(error: DugiteError): string {
       return 'A lock file already exists in the repository, which blocks this operation from completing.'
     case DugiteError.NoMergeToAbort:
       return 'There is no merge in progress, so there is nothing to abort.'
+    case DugiteError.NoExistingRemoteBranch:
+      return 'The remote branch does not exist.'
+    case DugiteError.LocalChangesOverwritten:
+      return 'Some of your changes would be overwritten.'
```

Comment:
<img width="447" alt="Screen Shot 2019-05-09 at 9 23 26 AM" src="https://user-images.githubusercontent.com/359239/57453189-22dca300-723c-11e9-896b-3fcca593bf53.png">

Can we make this message a bit more descriptive? Something like:

> Unable to switch branches as there are working directory changes that would be overwritten. Please commit or stash your changes.